src/Json.Schema.ToDotNet.UnitTests/DataModelGeneratorTests.cs

```diff
@@ -2363,7 +2369,152 @@ public int GetHashCode(C obj)
+        public int IntProp { get; set; }
+
+        /// <summary>
+        /// An integer property with a default value.
+        /// </summary>
+        [DataMember(Name = ""intPropWithDefault"", IsRequired = false, EmitDefaultValue = false)]
+        [DefaultValue(42)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public int IntPropWithDefault { get; set; }
+
+        /// <summary>
+        /// A number property.
+        /// </summary>
+        [DataMember(Name = ""numberProp"", IsRequired = false, EmitDefaultValue = false)]
+        public double NumberProp { get; set; }
+
+        /// <summary>
+        /// A number property with a default value.
+        /// </summary>
+        [DataMember(Name = ""numberPropWithDefault"", IsRequired = false, EmitDefaultValue = false)]
+        [DefaultValue(42.1)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public double NumberPropWithDefault { get; set; }
+
+        /// <summary>
+        /// A string property.
+        /// </summary>
+        [DataMember(Name = ""stringProp"", IsRequired = false, EmitDefaultValue = false)]
+        public string StringProp { get; set; }
+
+        /// <summary>
+        /// A string property with a default value.
+        /// </summary>
+        [DataMember(Name = ""stringPropWithDefault"", IsRequired = false, EmitDefaultValue = false)]
+        [DefaultValue(""42"")]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public string StringPropWithDefault { get; set; }
+
+        /// <summary>
+        /// A Boolean property.
+        /// </summary>
+        [DataMember(Name = ""boolProp"", IsRequired = false, EmitDefaultValue = false)]
+        public bool BoolProp { get; set; }
+
+        /// <summary>
+        /// A Boolean property with a true default value.
+        /// </summary>
+        [DataMember(Name = ""boolPropWithTrueDefault"", IsRequired = false, EmitDefaultValue = false)]
+        [DefaultValue(true)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public bool BoolPropWithTrueDefault { get; set; }
+
+        /// <summary>
+        /// A Boolean property with a false default value.
+        /// </summary>
+        [DataMember(Name = ""boolPropWithFalseDefault"", IsRequired = false, EmitDefaultValue = false)]
+        [DefaultValue(false)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public bool BoolPropWithFalseDefault { get; set; }
+    }
+}";
+
+            var generator = new DataModelGenerator(_settings, _testFileSystem.FileSystem);
+            JsonSchema schema = SchemaReader.ReadSchema(Schema, TestUtil.TestFilePath);
+
+            generator.Generate(schema);
+
+            var expectedContentsDictionary = new Dictionary<string, ExpectedContents>
```

Comment:
>expectedContentsDictionary [](start = 16, length = 26)

your fancy initialization here is working against readabilty, for those not familiar with the indexing syntax you're utilizing.

Here's another way to do it:

            var expectedContentsDictionary = new Dictionary<string, ExpectedContents>
            {
                { _settings.RootClassName,  new ExpectedContents {ClassContents = ExpectedClass } }
            };

Even putting your current code on one line helps communicate what's happening better, I think.

            var expectedContentsDictionary = new Dictionary<string, ExpectedContents>
            {
                [_settings.RootClassName] = new ExpectedContents { ClassContents = ExpectedClass }
            };

summary of note: improve readability if you can

Reply:
Putting it on one line is ok if there's only one initializer, but it spreads out too far when there is more than one:
```
            var expectedContentsDictionary = new Dictionary<string, ExpectedContents>
            {
                [_settings.RootClassName] = new ExpectedContents
                {
                    ClassContents = ExpectedRootClass,
                    ComparerClassContents = ExpectedRootComparerClass
                },
                ["Color"] = new ExpectedContents
                {
                    ClassContents = ExpectedColorClass,
                    ComparerClassContents = ExpectedColorComparerClass
                }
            };
```
IMO this is fine, and I'm not sympathetic to "hard to read if you don't know the language" ;-)

---
In reply to: [244509144](https://github.com/Microsoft/jschema/pull/62#discussion_r244509144) [](ancestors = 244509144)

Reply:
Code is written to serve the reader and not the author. Patterns at the leading edge of the language are often confusing to people. There often isn't common convention around how best to expresss them, as various codesmiths are tinkering. My note here is only that this fairly abstruse initialization syntax is not particularly readable. 

---
In reply to: [244511949](https://github.com/Microsoft/jschema/pull/62#discussion_r244511949) [](ancestors = 244511949,244509144)